### PR TITLE
Auto-take Profit changes information

### PR DESCRIPTION
**Pull request by piotrkonowrocki**
# Auto-take Profit changes information

All of the calculations are taken from multiply vault, but since they are trapped inside of a pipeline that doesn't allow any access from outside of it, I needed to take out some of the stuff out and create separated methods for it.

There are couple differences in changes information compared to figma designs, all of it were consulted:
- Numbers are displayed in full notations since we aren't using short notation anywhere,
- slippage, ETH price and price impact were removed,
- debt is displayed in DAI, not in $,
- estimated transaction costs are different (see: https://discord.com/channels/837076147694207067/839057646480785438/1024272474562633838).

I've also prepared `getEstimatedCostOnClose` thinking that it is going to be used in Stop-Loss and there is going to be new one, similar, `getEstimatedCostOnAdjustMultiply`.

image
  
## Changes 👷‍♀️

- Filled estimated profit with data calculated from UI state,
- filled changes infromation width data calculated from UI state,
- connected Auto-take Profit sidebar to missing generic methods for title, cta label etc.,
- created `estimatedCostOnClose` and `estimatedProfitOnClose` as boilerplate methods to be used in the rest of auto features.
  
## How to test 🧪

Try to add Auto-take Profit trigger and see how changes in form are reflected on summary.

### CODE REVIEW

**Review by adamskrodzki — DISMISSED** [review]

File: features/automation/common/estimations/estimatedCostOnClose.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  ethMarketPrice: BigNumber
+  lockedCollateral: BigNumber
+  toCollateral: boolean
+}) {
+  const { gasPrice$ } = useAppContext()
+  const [gasPrice] = useObservable(gasPrice$)
+
+  const currentDebt = debt.plus(debtOffset)
+  const toTokenAmount = lockedCollateral.times(colMarketPrice).times(one.minus(OAZO_FEE))
+  const requiredAmount = currentDebt.times(one.plus(OAZO_FEE)).times(one.plus(LOAN_FEE))
```
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply features/automation/common/estimations/estimatedCostOnClose.ts]: Irrelevant anymore, removed those calculations and taken them from `@oasisdex/multiply`.

File: features/automation/common/estimations/estimatedCostOnClose.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  const oasisFee = toCollateral
+    ? requiredAmount.times(OAZO_FEE)
+    : lockedCollateral.times(colMarketPrice).minus(toTokenAmount)
```
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply features/automation/common/estimations/estimatedCostOnClose.ts]: ditto

File: features/automation/common/estimations/estimatedCostOnClose.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  const loanFee = toCollateral ? requiredAmount.times(LOAN_FEE) : currentDebt.times(LOAN_FEE)
```
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply features/automation/common/estimations/estimatedCostOnClose.ts]: ditto

File: features/automation/common/estimations/estimatedProfitOnClose.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  debtOffset: BigNumber
+  lockedCollateral: BigNumber
+  toCollateral: boolean
+}) {
+  const { userSettings$ } = useAppContext()
+  const [userSettings] = useObservable(userSettings$)
+
+  const slippage = userSettings?.slippage || zero
+  const currentDebt = debt.plus(debtOffset)
+  const toTokenAmount = lockedCollateral.times(colMarketPrice).times(one.minus(OAZO_FEE))
```
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply features/automation/common/estimations/estimatedProfitOnClose.ts]: ditto

File: features/automation/common/estimations/estimatedProfitOnClose.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  const requiredAmount = currentDebt.times(one.plus(OAZO_FEE)).times(one.plus(LOAN_FEE))
+  const minToTokenAmount = toTokenAmount.times(one.minus(slippage))
```
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply features/automation/common/estimations/estimatedProfitOnClose.ts]: ditto

**Review by adamskrodzki — APPROVED** [review]

**Pull request merged**